### PR TITLE
Add ReasonML example

### PR DESCRIPTION
**Pull request by mike-engel**
This adds an example that co-mixes JavaScript (normal setup) with ReasonML react components using [ReasonReact](https://reasonml.github.io/reason-react) and [babel-plugin-bucklescript](https://github.com/mike-engel/babel-plugin-bucklescript).

This mimics the `shared-components` example, including styles. The only thing missing is the use of `Link`, which was causing much frustration during translation. If you're interested in this PR and want me to try harder to incorporate `Link`, let me know and I'll give it another go.

Finally, this *does* edit some global config (~`.gitignore` and~ `package.json`). If you're averse to those changes, let me know and I'll see what I can do about working around them. ~The `bs` (lib/bs) folder should be ignored as it includes build-time info (like `.next`).~

Thanks 🤗

### CODE REVIEW

**Review by timneutkens — APPROVED** [review]
Nice!

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # build output
 dist
 .next
+bs
+.merlin
```
Comment (timneutkens) [inline .gitignore]: Can there be in the gitignore of the example? or in the gitignore of the examples folder 👌

Comment (mike-engel) [inline .gitignore]: I went ahead and created on in this specific example's folder. Thanks!

Comment (timneutkens) [inline .gitignore]: ❤️ Thanks Mike

**Review by timneutkens — APPROVED** [review]
👍

**Pull request merged**